examples/src/main/scala/org/apache/spark/examples/ml/DataFrameExample.scala

```diff
@@ -54,14 +54,13 @@ object DataFrameExample {
       }
     }
 
-    parser.parse(args, defaultParams).map { params =>
-      run(params)
-    }.getOrElse {
-      sys.exit(1)
+    parser.parse(args, defaultParams) match {
+      case Some(params) => run(params)
+      case _ => sys.exit(1)
     }
```

Comment:
I wouldn't change these kinds of constructs.  It's almost purely a stylistic issue (for which I don't believe we have an officially declared preference), and 1) pattern matching over Options is poor form, IMO; 2) the pattern matching alternative here doesn't any more unambiguously imply or signal use of side effects.

If I had my way, we'd `fold` Options, but that's an argument I lost long ago:
```scala
val anOption: Option[T] = ...
anOption.fold { handleNone } { t => ... }
```